..
   Этот файл является частью проекта Logtalk <https://logtalk.org/>  
   Copyright 1998-2022 Paulo Moura <pmoura@logtalk.org>
   Идентификатор лицензии SPDX: Apache-2.0

   Лицензируется в соответствии с лицензией Apache, версия 2.0 (далее «Лицензия»);
   вы не можете использовать этот файл иначе, как в соответствии с Лицензией.
   Вы можете получить копию Лицензии на

       http://www.apache.org/licenses/LICENSE-2.0

   Если это не требуется законодательством или не согласовано в письменной форме, программное обеспечение
   распространяется по лицензии на условиях "КАК ЕСТЬ".,
   БЕЗ КАКИХ-ЛИБО ГАРАНТИЙ ИЛИ УСЛОВИЙ, явных или подразумеваемых.
   См. лицензию для конкретного языка, регулирующего разрешения и
   ограничения, предусмотренные Лицензией.


.. rst-class:: align-right

**built-in method**

.. index:: pair: representation_error/1; Built-in method
.. _methods_representation_error_1:

``representation_error/1``
==========================

Description
-----------

::

   representation_error(Flag)

Throws a representation error. Used when some representation limit is exceeded.
For example, trying to construct a compound term that exceeds the maximum arity
supported by the backend Prolog system. This built-in method is declared private
and thus cannot be used as a message to an object. Calling this predicate is
equivalent to the following sequence of goals:

::

   ...,
   context(Context),
   throw(error(representation_error(Flag), Context)).

This allows the user to generate errors in the same format used by the
runtime.

Possible values for ``Flag`` include:

- ``character``
- ``character_code``
- ``in_character_code``
- ``max_arity``
- ``max_integer``
- ``min_integer``
- ``lambda_parameters``
- ``entity_prefix``

Modes and number of proofs
--------------------------

::

   representation_error(+atom) - error

Errors
------

| When called:
|     ``representation_error(Flag)``

Examples
--------

::

   ...,
   Code > 127,
   representation_error(character_code).

.. seealso::

   :ref:`methods_catch_3`,
   :ref:`methods_throw_1`,
   :ref:`methods_context_1`,
   :ref:`methods_instantiation_error_0`,
   :ref:`methods_uninstantiation_error_1`,
   :ref:`methods_type_error_2`,
   :ref:`methods_domain_error_2`,
   :ref:`methods_existence_error_2`,
   :ref:`methods_permission_error_3`,
   :ref:`methods_evaluation_error_1`,
   :ref:`methods_resource_error_1`,
   :ref:`methods_syntax_error_1`,
   :ref:`methods_system_error_0`
